app/models/manageiq/providers/amazon/cloud_manager/refresh_worker.rb

```diff
@@ -1,3 +1,16 @@
 class ManageIQ::Providers::Amazon::CloudManager::RefreshWorker < ManageIQ::Providers::BaseManager::RefreshWorker
   require_nested :Runner
+
+  # overriding queue_name_for_ems so PerEmsWorkerMixin picks up *all* of the
+  # Amazon-manager types from here.
+  # This way, the refresher for Amazon's CloudManager will refresh *all*
+  # of the Amazon inventory across all managers.
+  def self.queue_name_for_ems(ems)
+    return ems unless ems.kind_of?(ExtManagementSystem)
+    ["ems_#{ems.id}"] + ems.child_managers.collect { |manager| "ems_#{manager.id}" }
```

Comment:
The parent implementation handles strange edge cases like if the input is a `Host`.

Maybe this should be:

```ruby
if ems.kind_of?(ExtManagementSystem)
  ["ems_#{ems.id}"] + ems.child_managers.collect { |manager| "ems_#{manager.id}" }
else
  super
end
```

Then, let `super` handle any of the edge cases.